src/Compilers/Core/CodeAnalysisTest/Diagnostics/SystemLanguageAnalyzer.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Semantics;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Diagnostics.SystemLanguage
+{
+    // These analyzers are not yet intended for any actual use. They exist solely to test IOperation support.
+
+    /// <summary>Analyzer used to test for loop IOperations.</summary>
+    public class FieldCouldBeReadOnlyAnalyzer : DiagnosticAnalyzer
+    {
+        private const string SystemCategory = "System";
+
+        public static readonly DiagnosticDescriptor FieldCouldBeReadOnlyDescriptor = new DiagnosticDescriptor(
+            "FieldCouldBeReadOnly",
+            "Field Could Be ReadOnly",
+            "Field is never modified and so could be readonly or const.",
+            SystemCategory,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        /// <summary>Gets the set of supported diagnostic descriptors from this analyzer.</summary>
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        {
+            get { return ImmutableArray.Create(FieldCouldBeReadOnlyDescriptor); }
+        }
+        
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(
+                 (compilationContext) =>
+                 {
+                     HashSet<IFieldSymbol> assignedToFields = new HashSet<IFieldSymbol>();
+                     HashSet<IFieldSymbol> mightBecomeReadOnlyFields = new HashSet<IFieldSymbol>();
+
+                     compilationContext.RegisterOperationBlockStartAction(
+                         (operationBlockContext) =>
+                         {
+                             IMethodSymbol containingMethod = operationBlockContext.OwningSymbol as IMethodSymbol;
+
+                             if (containingMethod != null)
+                             {
+                                 bool inConstructor = containingMethod.MethodKind == MethodKind.Constructor;
+                                 ITypeSymbol inStaticConstructor = containingMethod.MethodKind == MethodKind.StaticConstructor ? containingMethod.ContainingType : null;
```

Comment:
Nit, inStaticConstructor is badly named since it is a type. Did this use to be boolean?  Maybe 'staticConstructorType'?